eras/babbage/impl/testlib/Test/Cardano/Ledger/Babbage/ImpTest.hs

```diff
@@ -120,12 +125,37 @@ produceRefScripts ::
   NonEmpty (Script era) ->
   ImpTestM era (NonEmpty TxIn)
 produceRefScripts scripts = do
+  txId <- txIdTx <$> produceRefScriptsTx scripts
+  pure $ NE.zipWith (\_ -> mkTxInPartial txId) scripts (0 :| [1 ..])
+
+produceRefScriptsTx ::
+  (ShelleyEraImp era, BabbageEraTxOut era) =>
+  NonEmpty (Script era) ->
+  ImpTestM era (Tx era)
+produceRefScriptsTx scripts = do
   pp <- getsNES $ nesEsL . curPParamsEpochStateL
   txOuts <- forM scripts $ \script -> do
     addr <- freshKeyAddr_
     let txOutZero =
           mkBasicTxOut addr mempty & referenceScriptTxOutL .~ SJust script
     pure $ setMinCoinTxOut pp txOutZero
   let txBody = mkBasicTxBody & outputsTxBodyL .~ SSeq.fromList (NE.toList txOuts)
-  txId <- txIdTx <$> submitTx (mkBasicTx txBody)
-  pure $ NE.zipWith (\_ -> mkTxInPartial txId) scripts (0 :| [1 ..])
+  submitTx (mkBasicTx txBody)
+
+mkSpendUsingRefInputs ::
+  (ShelleyEraImp era, BabbageEraTxBody era) =>
+  TxIn ->
+  NonEmpty TxIn ->
+  Tx era
+mkSpendUsingRefInputs txIn refIns =
+  mkBasicTx $
+    mkBasicTxBody
+      & referenceInputsTxBodyL .~ Set.fromList (NE.toList refIns)
+      & inputsTxBodyL .~ [txIn]
+
+spendUsingRefInputs ::
```

Comment:
Same here:
```suggestion
submitTxWithRefInputs ::
```

Reply:
Indeed. Also more consistent with the rest of our `Imp` language. Thanks!